sys/include/net/credman.h

```diff
@@ -0,0 +1,192 @@
+/*
+ * Copyright (C) 2019 HAW Hamburg
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_credman (D)TLS Credential Manager
+ * @ingroup     net
+ * @brief       Credentials management module for (D)TLS
+ *
+ * @{
+ *
+ * @file
+ * @brief       (D)TLS credentials management module definitions
+ *
+ * @note        This module DOES NOT copies the credentials into the system. It
+ *              just holds the pointers to the credentials given by the user.
+ *              The user must make sure that these pointers are valid during the
+ *              lifetime of the application.
+ *
+ * @author      Aiman Ismail <muhammadaimanbin.ismail@haw-hamburg.de>
+ */
+
+#ifndef NET_CREDMAN_H
+#define NET_CREDMAN_H
+
+#include <unistd.h>
+#include <stdint.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief Maximum number of credentials in credential pool
+ */
+#ifndef CREDMAN_MAX_CREDENTIALS
+#define CREDMAN_MAX_CREDENTIALS  (2)
+#endif
+
+/**
+ * @brief Buffer of the credential
+ */
+typedef struct {
+    void *s;                /**< Pointer to the buffer */
+    size_t len;             /**< Length of s */
```

Comment:
```suggestion
    size_t len;             /**< Length of credman_buffer_t::s */
```

This way it can be parsed properly by `doxygen`